tasks/sign-binaries/sign-binaries.yaml

```diff
@@ -71,44 +71,239 @@ spec:
     - name: checksum-keytab-vol
       secret:
         secretName: $(params.checksumKeytab)
+    - name: mac-ssh-key-vol
+      secret:
+        secretName: $(params.macSSHKey)
   workspaces:
     - name: data
       description: Workspace to save the results to
   results:
+    - name: unsignedMacDigest
+      type: string
+      description: |
+        Digest used by Mac signing host to pull unsigned content via ORAS
+    - name: signedMacDigest
+      type: string
+      description: |
+        Digest used to pull signed Mac content back to pipeline via ORAS
     - name: unsignedWindowsDigest
       type: string
       description: |
-        Digest used by signing host to pull unsignged content via ORAS
+        Digest used by Windows signing host to pull unsigned content via ORAS
     - name: signedWindowsDigest
       type: string
       description: |
         Digest used to pull signed content back to pipeline via ORAS
     - name: unsignedMacDigest
       type: string
       description: |
-        Digest used by signing host to pull unsignged content via ORAS
+        Digest used by signing host to pull unsigned content via ORAS
     - name: signedMacDigest
       type: string
       description: |
         Digest used to pull signed content back to pipeline via ORAS
+    - name: binaries_path
+      type: string
+      description: |
+        Path where the final signed content is stored in the workspace
   steps:
     - name: push-unsigned-using-oras
       image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+      env:
+        - name: QUAY_USER
+          valueFrom:
+            secretKeyRef:
+              name: $(params.quaySecret)
+              key: username
+        - name: QUAY_PASS
+          valueFrom:
+            secretKeyRef:
+              name: $(params.quaySecret)
+              key: password
       script: |
         #!/usr/bin/env bash
+        set -eux
+
+        CONTENT_DIR=$(workspaces.data.path)/$(params.contentDir)
+        UNSIGNED_DIR=$CONTENT_DIR/unsigned
+        MAC_CONTENT=$UNSIGNED_DIR/macos
+        WINDOWS_CONTENT=$UNSIGNED_DIR/windows
+        LINUX_CONTENT=$CONTENT_DIR/linux
+
+        mkdir -p "$MAC_CONTENT" "$WINDOWS_CONTENT" "$LINUX_CONTENT"
+        cd "$CONTENT_DIR" || exit 1
+
+        # Loop through each gz file and move them to the appropriate directory
+        for file in *.gz; do
+          case "$file" in
+            (*darwin*)
+              mv "$file" unsigned/macos/
+              ;;
+            (*windows*)
+              mv "$file" unsigned/windows/
+              ;;
+            (*linux*)
+              mv "$file" linux/
+              ;;
+          esac
+        done
 
-        # TODO CLDX-134
+        # Unzip files in each directory
+        for dir in unsigned/macos unsigned/windows linux; do
+          for file in "$dir"/*.gz; do
+            gunzip "$file"
+          done
+        done
+
+        cd "$UNSIGNED_DIR" || exit 1
```

Comment:
Same comment as a few lines above, just opening a thread here too in case we have something better for it and don't forget this one